avaframe/com1DFAPy/SPHfunctionsCython.pyx

```diff
@@ -0,0 +1,1264 @@
+"""
+    function related to SPH calculations in com1DFA
+
+    to build: go to repository containing this file and run:
+    python setup.py build_ext --inplace
+"""
+
+# Load modules
+import copy
+import logging
+import math
+import numpy as np
+cimport numpy as np
+from libc cimport math as math
+# from libc.math cimport log as ln
+cimport cython
+
+# Local imports
+import avaframe.com1DFAPy.com1DFA as com1DFA
+import avaframe.com1DFAPy.DFAtools as DFAtls
+import avaframe.com1DFAPy.SPHfunctions as SPH
+import avaframe.in3Utils.geoTrans as geoTrans
+
+
+# create local logger
+# change log level in calling module to DEBUG to see log messages
+log = logging.getLogger(__name__)
+
+# how to compute SPH gradient:
+# 1) like in SAMOS AT just project on the plane
+# 2) project on the plane, compute the gradient in the local coord sys related
+# to the local plane and flow direction (orthogonal coord sys)
+# enables to choose earth pressure coefficients
+# 3) project on the plane, compute the gradient in the local coord sys related
+# to the local plane (tau1, tau2, n) non orthogonal coord sys
+cdef int SPHOption = 2
+
+cdef double rho = 200
+cdef double csz = 5
+cdef double gravAcc = 9.81
+
+ctypedef double dtypef_t
+ctypedef long dtypel_t
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def pointsToRasterC(x, y, z, Z0, csz=1, xllc=0, yllc=0):
```

Comment:
Is the C in all these names necessary? -> relates to my question whether it is a replacement or duplicate...

Reply:
You dont need the C in the name. This function does the same as pointtoRaster in geotrans but purely in C.
It is the case for almost all the NameOfFunctionC(), it is the C version of NameOfFunction. By choosing the flag flagCython you can decide to use cython functions or numpy ones